Cargo.toml

```diff
@@ -40,10 +40,13 @@ quickcheck = { version = "0.4.1", default-features = false }
 rand = "0.3.15"
 
 [features]
+default = ["use_std"]
 # Enable to use the unstable pattern traits defined in std.
 pattern = []
 # Enable to use simd acceleration.
 simd-accel = ["simd"]
+# Allow compilation of dependencies without std
+use_std = ["aho-corasick/use_std", "memchr/use_std"]
```

Comment:
So now that I'm thinking about this (and this comment applies to the `aho-corasick` crate as well), I'm thinking that `use_std` might not be an appropriate name. It's surely appropriate for `memchr` since the absence of `use_std` actually means that `memchr` won't use `std` at all. But in the case of `aho-corasick` and `regex`, that's not what `use_std` means at all. Instead the absence of `use_std` in `regex` and `aho-corasick` seems to mean that it won't explicitly import from `libc` anywhere, which is quite a bit different than not using `std`.

I wonder if it makes sense to add a `use_libc` feature instead (and similarly for `aho-corasick`). That would then in turn have to imply the `memchr/use_std` feature.